batcher/aligned-sdk/src/sdk.rs

```diff
@@ -655,6 +660,81 @@ pub async fn get_balance_in_aligned(
     }
 }
 
+/// Saves AlignedVerificationData in a file.
+/// # Arguments
+/// * `batch_inclusion_data_directory_path` - The path of the directory where the data will be saved.
+/// * `aligned_verification_data` - The aligned verification data to be saved.
+/// # Returns
+/// * Ok if the data is saved successfully.
+/// # Errors
+/// * `FileError` if there is an error writing the data to the file.
+pub fn save_response(
+    batch_inclusion_data_directory_path: PathBuf,
+    aligned_verification_data: &AlignedVerificationData,
+) -> Result<(), errors::FileError> {
+    let _ = save_response_cbor(
+        batch_inclusion_data_directory_path.clone(),
+        &aligned_verification_data.clone(),
+    )?;
+    let _ = save_response_json(
+        batch_inclusion_data_directory_path,
+        &aligned_verification_data,
+    )?;
+    Ok(())
+}
+fn save_response_cbor(
+    batch_inclusion_data_directory_path: PathBuf,
+    aligned_verification_data: &AlignedVerificationData,
+) -> Result<(), errors::FileError> {
+    let batch_merkle_root = &hex::encode(aligned_verification_data.batch_merkle_root)[..8];
+    let batch_inclusion_data_file_name = batch_merkle_root.to_owned()
+        + "_"
+        + &aligned_verification_data.index_in_batch.to_string()
+        + ".cbor";
+
+    let batch_inclusion_data_path =
+        batch_inclusion_data_directory_path.join(batch_inclusion_data_file_name);
+
+    let data = cbor_serialize(&aligned_verification_data)?;
+
+    let mut file = File::create(&batch_inclusion_data_path)
+        .map_err(|e| errors::FileError::IoError(batch_inclusion_data_path.clone(), e))?;
+    file.write_all(data.as_slice())
+        .map_err(|e| errors::FileError::IoError(batch_inclusion_data_path.clone(), e))?;
+    info!(
+        "Batch inclusion data written into {}",
+        batch_inclusion_data_path.display()
+    );
+
+    Ok(())
+}
+fn save_response_json(
+    batch_inclusion_data_directory_path: PathBuf,
+    aligned_verification_data: &AlignedVerificationData,
+) -> Result<(), errors::FileError> {
+    let batch_merkle_root = &hex::encode(aligned_verification_data.batch_merkle_root)[..8];
+    let batch_inclusion_data_file_name = batch_merkle_root.to_owned()
+        + "_"
+        + &aligned_verification_data.index_in_batch.to_string()
+        + ".json";
+
+    let batch_inclusion_data_path =
+        batch_inclusion_data_directory_path.join(batch_inclusion_data_file_name);
+
+    let data = serde_json::to_vec(&aligned_verification_data).unwrap();
```

Comment:
Just out of curiosity, will this still saving the bytes directly into the json file? Wouldn't be good to add the encoding to hex before saving to make it easier to use?

Reply:
This has shown me the `proof_commitment` stored is different than the `proof_hash` shown in the explorer. I am searching why and how to fix

Reply:
Solved! Explorer was using Sha3 instead of Keccak (I thought they were always interchangeable, they are not.)

Reply:
Excellent!